app/Console/Commands/DropUpload.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\ImportMeta;
+use App\Models\Mismatch;
+use Illuminate\Console\Command;
+
+class DropUpload extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'uploads:drop {id : ID of the upload to drop}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Drop an uploaded mismatch file by ID.';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle()
+    {
+        $import = ImportMeta::find($this->argument('id'));
+        if (!$import) {
+            $this->error(__('admin.dropUpload:notFound', ['id' => $this->argument('id')]));
+
+            return 1;
+        }
+
+        $mismatchesToDrop = Mismatch::where('import_id', '=', $import->id)->count();
+        $this->line(__(
+            'admin.dropUpload:dropping',
+            [
+                'id' => $import->id,
+                'mismatches' => $mismatchesToDrop
+            ]
+        ));
+
+        if ($this->confirm('Are you sure?')) {
+            Mismatch::where('import_id', '=', $import->id)->delete();
+            $import->delete();
```

Comment:
I think it will be best if we put these deletions inside a [database transaction](https://laravel.com/docs/8.x/database#database-transactions). That way, the db will roll back in case something catastrophic happens while deleting.

Reply:
Yes, done. I didn't add any explicit error handling, I guess in case of catastrophy the command will just throw and exit with an error? Or does the success message belong in the `DB::transaction` block?

Reply:
It'll throw, so I think it's okay